sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -557,6 +558,8 @@ def __init__(self, options, root_staging_location=None):
     if self.google_cloud_options.no_auth:
       credentials = None
     else:
+      set_impersonation_accounts(
+          self.google_cloud_options.impersonate_service_account)
```

Comment:
Around here would be the other use of `get_credentials(gcp_options)`. Perhaps that already lives in `get_service_credentials`. And around here is where you would produce the modified set of pipeline options for the job submission. I presume they are pickled into the job somewhere hereabouts?

Reply:
yes, I found a point where the pipeline options are modified for job submission and modified it so that impersonation doesn't propegate to the workers.

For anyone unfamiliar looking at this comment, the service worker that runs jobs in dataflow is unique and we don't want to try to do impersonation on the service worker, but rather only on the submission to dataflow.